health_check_storage_s3/plugin_health_check.py

```diff
@@ -0,0 +1,27 @@
+# -*- coding: utf-8 -*-
+import logging
+
+from health_check.plugins import plugin_dir
+from health_check_storage.base import StorageHealthCheck
+
+
+class S3BotoStorageHealthCheck(StorageHealthCheck):
+    """
+    Tests the status of a `S3BotoStorage` file storage backend.
+
+    S3BotoStorage is included in the `django-storages` package
+    and recommended by for example Amazon and Heroku for Django
+    static and media file storage on cloud platforms.
+
+    django-storages can be found at https://git.io/v1lGx
+    S3BotoStorage can be found at https://git.io/v1lGF
+    """
+
+    logger = logging.getLogger(__name__)
+    storage = 'storages.backends.s3boto.S3BotoStorage'
+
+    def check_delete(self, file_name):
+        storage = self.get_storage()
+        storage.delete(file_name)
```

Comment:
The S3 storage backend doesn't actually return the deleted item marker or state and the delete operation is asynchronous. The deletion state of the file can not be checked with the current S3BotoStorage storage API. This is the original reason for writing deletion here this way. The old implementation checks if the file has been deleted, but that always raises because the S3 bucket is not in sync yet.

This is due to limitations in the S3BotoStorage interface. I am going to ask for a refactor in that package as well, but that will be a breaking API change, whereas this one should not be.

Possible ways to implement checking if deletion works are:

- Wait for a small while before checking, trying delays between e.g. in 300-3000ms range. This could return false positives due to async calls;
- Implement locks and only allow one check to run here per time. Check if there is more than one file to be deleted. This should work but limits the health checker rate severely.
- Implement the deletion state checker independently of the S3BotoStorage API, fetch delete markers and compare them. This is an extremely volatile approach which is tied to the Boto API version.

So, really, there is no clean way for doing this yet.